flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/api/internal/TableEnvironmentImpl.java

```diff
@@ -290,6 +292,40 @@ public String explain(boolean extended) {
 		return planner.getCompletionHints(statement, position);
 	}
 
+	@Override
+	public Table sql(String statement) {
+		List<Operation> operations = planner.parse(statement);
+		Preconditions.checkState(operations.size() <= 1,
```

Comment:
change to operations.size() == 1? Otherwise it will be unsafe for your next line: `Operation operation = operations.get(0);`